### PR TITLE
Adding R support

### PR DESCRIPTION
**Pull request by anneschuth**
## Description

Adding support for Models written in R. 

## Motivation and Context

Many models we encounter in the wild are written in R.

## How to Test

[Provide clear instructions on how to test and verify the changes introduced by this pull request, including any specific unit tests you have created to demonstrate your changes.]

## Checklist

Please check all the boxes that apply to this pull request using "x":

- [ ]  I have tested the changes locally and verified that they work as expected.
- [ ]  I have added or updated the necessary documentation (README, API docs, etc.).
- [ ]  I have added appropriate unit tests or functional tests for the changes made.
- [ ]  I have followed the project's coding conventions and style guidelines.
- [ ]  I have rebased my branch onto the latest commit of the main branch.
- [ ]  I have squashed or reorganized my commits into logical units.
- [ ]  I have added any necessary dependencies or packages to the project's build configuration.
- [ ]  I have performed a self-review of my own code.
- [ ]  I have read, understood and agree to the Developer Certificate of Origin below, which this project utilises.

## Screenshots (if applicable)

[If the changes involve visual modifications, include screenshots or GIFs that demonstrate the changes.]

## Additional Notes

[Add any additional information or context that might be relevant to reviewers.]

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>

### CODE REVIEW

File: test-engine-core-modules/requirements.txt [hunk 1]
```diff
@@ -13,7 +13,7 @@ h5py==3.9.0 ; python_version >= "3.10" and python_version < "3.12"
 idna==3.4 ; python_version >= "3.10" and python_version < "3.12"
 jax==0.4.13 ; python_version >= "3.10" and python_version < "3.12"
 joblib==1.2.0 ; python_version >= "3.10" and python_version < "3.12"
-keras==2.12.0 ; python_version >= "3.10" and python_version < "3.12"
```
Comment (anneschuth) [inline test-engine-core-modules/requirements.txt:16]: I'll probably need to drop bumping of Tensorflow/Keras from this PR, or create a new one with this specific change. 

The change is in here to make tests run on my MacBook.

File: test-engine-core/test_engine_core/plugins/plugins_manager.py [hunk 2]
```diff
@@ -422,29 +423,23 @@ def _sort_plugin(plugin_tuple: Tuple[str, IPlugin]) -> int:
...
+            prio_type = plugin_tuple[1].Plugin.get_serializer_plugin_type()
+            priority = PluginManager._serializer_priority_list
         else:
-            return list(
-                PluginManager._plugins[PluginType.ALGORITHM.name].values()
-            ).index(plugin_tuple[1])
+            prio_type = plugin_tuple[1]
+            priority = PluginManager._plugins[PluginType.ALGORITHM.name].values()
+
+        return priority.index(prio_type) if prio_type in priority else len(priority)
```
Comment (anneschuth) [inline test-engine-core/test_engine_core/plugins/plugins_manager.py:441]: Curious what you think, took me a while to find why my new model wouldn't be incorporated. This change allows for adding models without specifying their priority (putting them last).

**Comment by miyamaya9** [comment]
Hi @anneschuth our developers will be looking into it and get back to you again. Meanwhile could you provide us the unit test. Thank you.

**Comment by anneschuth** [comment]
This is a draft PR still, I'm happy to close it until I finish it up if it
is confusing to have it here?

**Comment by imda-benedictlee** [comment]
Hi @anneschuth, no worries. You can leave the PR as draft. When you are ready, do let me or @kimeetok know and we will review the PR. Thanks for your contribution!

**Comment by anneschuth** [comment]
@imda-benedictlee @kimeetok we've created a new PR in #269 in favour of this one, could you review that one?

**Pull request closed without merging**